src/main/java/com/sumologic/log4j/http/SumoHttpSender.java

```diff
@@ -25,136 +25,121 @@
-
-    public void setUrl(String url) {
-        this.url = url;
-    }
-
-    public void setConnectionTimeout(int connectionTimeout) {
-        this.connectionTimeout = connectionTimeout;
-    }
-
-    public void setSocketTimeout(int socketTimeout) {
-        this.socketTimeout = socketTimeout;
-    }
 
-    public boolean isInitialized() {
-        return httpClient != null;
-    }
-
-    public void init() {
-        RequestConfig requestConfig = RequestConfig.custom()
-                .setSocketTimeout(socketTimeout)
-                .setConnectTimeout(connectionTimeout)
-                .build();
-
-        HttpClientBuilder builder = HttpClients.custom()
-                .setConnectionManager(new PoolingHttpClientConnectionManager())
-                .setDefaultRequestConfig(requestConfig);
-
-        HttpProxySettingsCreator creator = new HttpProxySettingsCreator(proxySettings);
-        creator.configureProxySettings(builder);
-
-        httpClient = builder.build();
-    }
-
-    public void close() throws IOException {
-        httpClient.close();
-        httpClient = null;
-    }
-
-    public void send(String body, String name) {
-        keepTrying(body, name);
-    }
-
-    private void keepTrying(String body, String name) {
-        boolean success = false;
-        do {
-            try {
-                trySend(body, name);
-                success = true;
-            } catch (Exception e) {
-                try {
-                    Thread.sleep(retryInterval);
-                } catch (InterruptedException e1) {
-                    break;
-                }
-            }
-        } while (!success && !Thread.currentThread().isInterrupted());
-    }
-
-    private void trySend(String body, String name) throws IOException {
-        HttpPost post = null;
+public class SumoHttpSender
+{
+  private static final Logger logger = StatusLogger.getLogger();
+
+  private final long retryInterval;
+  private final String url;
+  private final CloseableHttpClient httpClient;
+  private final String name;
+  private final String host;
+  private final String category;
+
+  public SumoHttpSender(
+      long retryInterval,
+      String url,
+      CloseableHttpClient httpClient,
+      String name,
+      String host,
+      String category
+  )
+  {
+    this.retryInterval = retryInterval;
+    this.url = url;
+    this.httpClient = httpClient;
+    this.name = name;
+    this.host = host;
+    this.category = category;
+  }
+
+  public boolean isInitialized()
+  {
+    return httpClient != null;
+  }
+
+  public void send(byte[] body)
+  {
+    keepTrying(body);
+  }
+
+  private void keepTrying(byte[] body)
+  {
+    boolean success = false;
+    int nTry = 1;
+    do {
+      try {
+        trySend(body);
+        success = true;
+      }
+      catch (Exception e) {
+        // Exponential backoff with jitter
+        final double fuzzyMultiplier = Math.min(Math.max(1 + 0.2 * ThreadLocalRandom.current().nextGaussian(), 0), 2);
```

Comment:
Please extract this as a method and leave a comment pointing to the original source